core/vm/logger.go

```diff
@@ -225,9 +257,29 @@ func (l *StructLogger) CaptureEnd(output []byte, gasUsed uint64, t time.Duration
 	}
 }
 
+func (l *StructLogger) CaptureEnter(typ OpCode, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int) {
+	in := make([]byte, len(input))
+	copy(in, input)
+	// TODO: should we honor `l.Cfg.Limit` for frames too?
```

Comment:
Nah, I don't think that's needed. You can only do so many scope transition on sane gas levels